### PR TITLE
Gb/pysam2

### PR DESCRIPTION
**Pull request by grantbuster**

### CODE REVIEW

**Comment by grantbuster** [comment]
Alright @MRossol, I think we're good to go on pysam2!

Major changes:
- PySAM version checker
- Version checker will update deprecated SAM config keys (wind losses is only current deprecated key)
- Refactor of the RevPySam structure including renaming of classes (annoying but more accurate) and dedicated defaults module
- Refactor of reV-SAM technology strings (must be explicit pvwattsv5, pvwattsv7, windpower, tcsmoltensalt)
- Additional tests for wind losses (with icing and low temp cutoff)
- Additional test for PVWattsv7 (simple CF mean baseline test)
- Additional tests for PySAM default objects (simple annual energy production baseline tests for fast running modules)

File: reV/SAM/__init__.py [hunk 1]
```diff
@@ -3,5 +3,5 @@
 reV-SAM interface module
 """
 from .econ import LCOE, SingleOwner
-from .generation import PV, CSP, LandBasedWind, OffshoreWind
+from .generation import Pvwattsv5, Pvwattsv7, TcsMoltenSalt, WindPower
```
[the remark this answers is not in the document]
Comment (grantbuster) [reply reV/SAM/__init__.py:6]: I think we should just use TcsMoltenSalt (which CSP was interpreted as). tcsmoltensalt is the SAM module. Thoughts?

File: reV/SAM/generation.py [hunk 2]
```diff
@@ -484,6 +488,11 @@ def set_nsrdb(self, resource):
         resource['lon'] = self.meta['longitude']
         resource['tz'] = self.meta['timezone']
 
+        if 'elevation' in self.meta:
+            resource['elev'] = self.meta['elevation']
+        else:
+            resource['elev'] = 0.0
+
         resource['minute'] = self.ensure_res_len(time_index.minute)
```
[the remark this answers is not in the document]
Comment (grantbuster) [reply reV/SAM/generation.py:496]: Work smart not hard, i like it

File: reV/handlers/sam_resource.py [hunk 3]
```diff
@@ -0,0 +1,569 @@
...
+import logging
+
+from reV.handlers.parse_keys import parse_keys
+from reV.utilities.exceptions import (HandlerKeyError, HandlerRuntimeError,
+                                      HandlerValueError, SAMInputWarning)
+
+logger = logging.getLogger(__name__)
+
+
+class SAMResource:
```
[the remark this answers is not in the document]
Comment (grantbuster) [reply reV/handlers/sam_resource.py:18]: Yes. I made modifications to it ages ago, I think the median thing was the only thing that didnt get passed through to rex.

[the remark this answers is not in the document]
Comment (grantbuster) [reply reV/handlers/sam_resource.py:18]: I guess the file got passed through during the rebases. will remove.

File: conda.recipe/meta.yaml [hunk 4]
```diff
@@ -16,7 +16,7 @@ requirements:
     - python
     - click
     - h5py
-    - nrel-pysam==1.2.1
+    - nrel-pysam>=2.1.1
```
Comment (grantbuster) [inline conda.recipe/meta.yaml:19]: @MRossol, im assuming this is all I need to do to update the Conda env for pysam2?

Comment (MRossol) [reply conda.recipe/meta.yaml:19]: yup!

**Pull request merged**